content/docs/setup/terraform/_index.md

```diff
@@ -6,52 +6,274 @@ weight = 30
+Worker to preform sequentially. This template contains a single task with a
+single action, which is to perform [“hello-world”](/docs/examples/hello-world). Just as in the hello-world
+example, the “hello-world” image doesn’t contain any instructions that the
+Worker will perform. It is just a placeholder in the template so a workflow can
+be created and pushed to the Worker.
+
+```
+cat > hello-world.yml  <<EOF
+version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+  - name: "hello world"
+    worker: "{{.device_1}}"
+    actions:
+      - name: "hello_world"
+        image: hello-world
+        timeout: 60
+EOF
+```
+
+Create the template and push it to the database with the `tink template create` command.
+
+```
+$ docker exec -i deploy_tink-cli_1 tink template create --name hello-world < ./hello-world.yml
+
+Created Template:  75ab8483-6f42-42a9-a80d-a9f6196130df
+```
 
-If you do not wish to use Terraform, you can provision the servers manually with the following configurations.
+## Create a workflow
 
-#### Provisioner
+The next step is to combine both the hardware data and the template to create a workflow.
 
-- Plan: c3.small.x86 (or any plan that supports Layer 2)
-- OS: Ubuntu 18.04 LTS
-- After device is provisioned:
-  - Convert network type to Mixed/Hybrid
-  - Attach VLAN to interface eth1 (under Layer 2)
+- First, the workflow needs to know which template to execute. The Template ID you should use was returned by `tink template create` command executed above.
+- Second, the Workflow needs a target, defined by the hardware data. In this example, the target is identified by the MAC address you got back from the `terraform apply` command
 
-#### Worker
+Combine these two pieces of information and create the workflow with the `tink workflow create` command.
 
-- Facility: <same_as_provisioner>
-- Plan: c3.small.x86 (or any plan that supports Layer 2)
-- OS: Custom iPXE
-  - IPXE Script URL: https://boot.netboot.xyz
-  - Always/Persist PXE: true
-- After device is provisioned:
-  - Convert network type to Layer 2 (individual)
-  - Attach VLAN to interface eth0
-  - Same VLAN as provisioner
+```
+$ docker exec -i deploy_tink-cli_1 tink workflow create \
+    -t 75ab8483-6f42-42a9-a80d-a9f6196130df \
+    -r '{"device_1":"1c:34:da:5c:36:88"}'
+
+Created Workflow:  a8984b09-566d-47ba-b6c5-fbe482d8ad7f
+```
+
+The command returns a Workflow ID and if you are watching the logs, you will see:
+
+```
+tink-server_1  | {"level":"info","ts":1592936829.6773047,"caller":"grpc-server/workflow.go:63","msg":"done creating a new workflow","service":"github.com/tinkerbell/tink"}
+```
+
+## Check workflow status
+
+You can not ssh into the worker but you can use the CLI from the provisioner to
+validate if the workflow completed correctly using the `tink workflow events`
+command:
+
+```
+docker exec -i deploy_tink-cli_1 tink workflow events a8984b09-566d-47ba-b6c5-fbe482d8ad7f
```

Comment:
I did not see the event fire when I tested. :-(

Reply:
It takes time but I will have a look :)